servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -569,6 +617,33 @@ public long createFile(TachyonURI path, long blockSizeBytes, boolean recursive,
     if (mWhitelist.inList(path.toString())) {
       inode.setCacheable(true);
     }
+
+    if (ttl != Constants.NO_TTL) {
```

Comment:
To make code more isolated, you can create a class `TTLBucketList` 
line 620-644 become the insert method of this `TTLBucketList`,
line 1425 and below can be the update or some getter method of `TTLBucketList` which removes the head bucket to start TTLChecker service.